File: ALGORITHMS/PYTHON/grades.py
def gradingStudents(grades):
    ans = []
    for i in range(len(grades)):
        next = grades[i]
        if grades[i] >= 38:
            while next % 5 !=0:
                next = next + 1
            if next - grades[i] < 3:
                ans.append(next)
            else:
                ans.append(grades[i])
        else:
            ans.append(grades[i])

    return ans
         





print (gradingStudents([73,67,38,33]))

File: Extra/JAVA/grades.java
public class grades{
    static int[] gradingStudents(int[] grades) {
        int  calif [] = new int[grades.length];
        int aux;

        for (int i = 0; i < grades.length; i++){
            aux = grades[0];
            if (grades[i] >=0){
                while (aux % 5 != 0){
                    aux = aux + 1;
                }
                if (aux - grades[i] < 3){
                    calif[i] = aux;
                }
                else{
                    calif[i] = grades[i];
                }

            }
            else{
                calif[i] = grades[i];
            }
        }

        return calif;

    }


    public static void mian(String[] args){
        int [] grade = {73,67,38,33}; 
        System.out.println(gradingStudents(grade));
    }
}